src/components/transactions/FlowCommons/TxModalDetails.tsx

```diff
@@ -98,42 +115,58 @@ export const DetailsNumberLineWithSub = ({
   futureValueUSD,
   hideSymbolSuffix,
   color,
+  tokenIcon,
+  loading = false,
 }: DetailsNumberLineWithSubProps) => {
   return (
     <Row caption={description} captionVariant="description" mb={4} align="flex-start">
       <Box sx={{ display: 'flex', flexDirection: 'column', alignItems: 'flex-end' }}>
-        <Box sx={{ display: 'flex', alignItems: 'center' }}>
-          {value && (
-            <>
-              <FormattedNumber value={value} variant="secondary14" color={color} />
-              {!hideSymbolSuffix && (
-                <Typography ml={1} variant="secondary14">
-                  {symbol}
-                </Typography>
-              )}
-              <SvgIcon color="primary" sx={{ fontSize: '14px', mx: 1 }}>
-                <ArrowNarrowRightIcon />
-              </SvgIcon>
-            </>
-          )}
-          <FormattedNumber value={futureValue} variant="secondary14" color={color} />
-          {!hideSymbolSuffix && (
-            <Typography ml={1} variant="secondary14">
-              {symbol}
-            </Typography>
-          )}
-        </Box>
-        <Box sx={{ display: 'flex', alignItems: 'center' }}>
-          {valueUSD && (
-            <>
-              <FormattedNumber value={valueUSD} variant="helperText" compact symbol="USD" />
-              <SvgIcon color="primary" sx={{ fontSize: '14px', mx: 1 }}>
-                <ArrowNarrowRightIcon />
-              </SvgIcon>
-            </>
-          )}
-          <FormattedNumber value={futureValueUSD} variant="helperText" compact symbol="USD" />
-        </Box>
+        {loading ? (
+          <Skeleton variant="rectangular" height={20} width={100} sx={{ borderRadius: '4px' }} />
+        ) : (
+          <Box sx={{ display: 'flex', alignItems: 'center' }}>
+            {value && (
+              <>
+                <FormattedNumber value={value} variant="secondary14" color={color} />
+                {!hideSymbolSuffix && (
+                  <Typography ml={1} variant="secondary14">
+                    {symbol}
+                  </Typography>
+                )}
+                <SvgIcon color="primary" sx={{ fontSize: '14px', mx: 1 }}>
+                  <ArrowNarrowRightIcon />
+                </SvgIcon>
+              </>
+            )}
+            {tokenIcon && <TokenIcon symbol={tokenIcon} sx={{ mr: 1, fontSize: '14px' }} />}
+            <FormattedNumber value={futureValue} variant="secondary14" color={color} />
+            {!hideSymbolSuffix && (
+              <Typography ml={1} variant="secondary14">
+                {symbol}
+              </Typography>
+            )}
+          </Box>
+        )}
+        {loading ? (
+          <Skeleton
+            variant="rectangular"
+            height={15}
+            width={80}
+            sx={{ borderRadius: '4px', marginTop: '4px' }}
+          />
+        ) : (
```

Comment:
Would it be easier to read this as one ternary?

`{loading ? <><Skelaton /><Skelaton /></> : <><Box>...</Box><Box>...</Box></>}`

Reply:
Not sure I'm following what you mean here. It's already using a ternary.

Reply:
Instead of two separate ternaries for loading and a box in the jsx markup, combining them into a singular one may be less verbose

Reply:
gotcha, yep updated here and in `SwapModalDetails`